packages/dds/tree/src/test/utils.ts

```diff
@@ -1300,3 +1302,23 @@ export function moveWithin(
 ) {
 	editor.move(field, sourceIndex, count, field, destIndex);
 }
+
+/**
+ * Clones a batch of revertibles for a target branch.
+ * @param revertibles - Array of revertibles to clone
+ * @param targetBranch - The target branch to clone the revertibles for
+ * @returns Array of cloned revertibles, maintaining the same order as the input
+ */
+export function cloneRevertiblesInBatch(
+	revertibles: RevertibleAlpha[],
+	targetBranch: TreeBranch,
+): RevertibleAlpha[] {
+	const disposedRevertible = revertibles.find(
+		(revertible) => revertible.status === RevertibleStatus.Disposed,
+	);
```

Comment:
nitpick: You can reduce the runtime complexity from 2n to n by doing something like
```ts
const clonedRevertibles = [];
for (const revertible of revertibles) {
  if (revertible.status === RevertibleStatus.Disposed) {
    throw new Error("List of revertible should not contain disposed revertibles.");
  }
  clonedRevertibles.push(revertible.clone(targetBranch));
} 

return clonedRevertibles;
```

Reply:
I should practice some LCs again 💦 . Thanks!

Reply:
I think what you have currently is better. If any revertibles further in the list are already disposed, this change would mean that we did extra work cloning the previous ones. We'd also have to deal with explicitly disposing the cloned ones to make sure we're not holding onto any resources

Reply:
+1 to Jenn, I didn't think about the complexity of the clone operation itself. Turns out you had the right way to begin with Ji!

Reply:
Thanks guys for the feedback!